standalone-metastore/metastore-rest-catalog/src/main/java/org/apache/iceberg/rest/HMSCatalogFactory.java

```diff
@@ -35,19 +38,33 @@
  * Catalog &amp; servlet factory.
  */
 public class HMSCatalogFactory {
-  private static final String SERVLET_ID_KEY = "metastore.in.test.iceberg.catalog.servlet.id";
+ private static final String SERVLET_ID_KEY = "metastore.in.test.iceberg.catalog.servlet.id";
 
-  private final Configuration configuration;
-  private final int port;
-  private final String path;
+  /**
+   * Convenience soft reference to last catalog.
+   */
+  protected static final AtomicReference<Reference<Catalog>> catalogRef = new AtomicReference<>();
```

Comment:
remove it

Reply:
I've removed this and changed the logic in the latest iteration to check whether a cached table still points to the most up to date location (reusing @okumin idea).